### PR TITLE
Containers: skip rpm backend check for <15-SP3

### PR DESCRIPTION
**Pull request by jlausuch**
This fixes [this](https://openqa.suse.de/tests/5959678#step/docker_image/99) problem

### CODE REVIEW

File: lib/containers/common.pm [hunk 1]
```diff
@@ -253,22 +253,20 @@ sub scc_restore_docker_image_credentials {
...
     die 'Argument $runtime not provided!' unless $runtime;
 
-    my $backend = script_output "$runtime run $image rpm --eval %_db_backend";
     my ($running_version, $sp, $host_distri) = get_os_release("$runtime run $image");
-    record_info('RPM check', "The rpm db backend in the image $image is $backend");
     # TW and SLE 15-SP3+ uses rpm-ndb in the image
     if ($host_distri eq 'opensuse-tumbleweed' || ($host_distri eq 'sles' && check_version('>=15-SP3', "$running_version-SP$sp", qr/\d{2}(?:-sp\d)?/))) {
-        $expected = 'ndb';
+        record_info('RPM check', "The rpm db backend in the image $image is $backend");
+        my $backend = script_output "$runtime run $image rpm --eval %_db_backend";
```
Comment (Vogtinator) [inline lib/containers/common.pm:267]: Why not just `validate_script_output`?

**Comment by jlausuch** [comment]
VRs:
[SLE 15-SP3](http://fromm.arch.suse.de/tests/1148)
[SLE 15-SP3 on Centos](http://fromm.arch.suse.de/tests/1147)
[SLE 15-SP2](http://fromm.arch.suse.de/tests/1150)
[SLE 12-SP5](http://fromm.arch.suse.de/tests/1151)
[TW](http://fromm.arch.suse.de/tests/1152)
[Leap 15.3](http://fromm.arch.suse.de/tests/1153)

**Review by Vogtinator — APPROVED** [review]

File: lib/containers/common.pm [hunk 2]
```diff
@@ -253,22 +253,18 @@ sub scc_restore_docker_image_credentials {
...
+    my $runtime = $args{runtime};
 
     die 'Argument $image not provided!'   unless $image;
     die 'Argument $runtime not provided!' unless $runtime;
 
-    my $backend = script_output "$runtime run $image rpm --eval %_db_backend";
     my ($running_version, $sp, $host_distri) = get_os_release("$runtime run $image");
-    record_info('RPM check', "The rpm db backend in the image $image is $backend");
     # TW and SLE 15-SP3+ uses rpm-ndb in the image
     if ($host_distri eq 'opensuse-tumbleweed' || ($host_distri eq 'sles' && check_version('>=15-SP3', "$running_version-SP$sp", qr/\d{2}(?:-sp\d)?/))) {
```
Comment (jlausuch) [inline lib/containers/common.pm:265]: @Vogtinator will ndb will be in 15-SP1 and 15-SP2 in the future?

Comment (Vogtinator) [reply lib/containers/common.pm:265]: Unlikely FWICT.

Comment (b10n1k) [reply lib/containers/common.pm:265]: https://jira.suse.com/browse/SLE-7272 looks like a ticket in progress for sle15sp2

**Review by b10n1k — COMMENTED** [review]
IMO it should be checked against all the distros. it s up to you.

File: lib/containers/common.pm (hunk 2 above)
Comment (Vogtinator) [reply lib/containers/common.pm:265]: https://jira.suse.com/browse/ECO-2409 is the main part. It's waiting for an RPM update which will probably never happen.

**Comment by jlausuch** [comment]
> IMO it should be checked against all the distros. it s up to you.

I would say only where the feature has been introduced, cause it comes from a Jira feature for specific product version. @Vogtinator wdyt?

File: lib/containers/common.pm (hunk 2 above)
Comment (jlausuch) [reply lib/containers/common.pm:265]: And if it happens, we will be simple need to update the condition to `>=15-SP2`

**Review by b10n1k — APPROVED** [review]
+1

**Review by pdostal — APPROVED** [review]

**Pull request merged**